tsp-output/@azure-tools/typespec-csharp/azoai/AzureOpenAI/Models/AzureModelExtensions.cs

```diff
@@ -0,0 +1,133 @@
+﻿using OpenAI.Models;
+using System.Diagnostics;
+using System.Text.Json;
+
+namespace AzureOpenAI.Models;
+
+public static class AzureModelExtensions
+{
+    public static IList<AzureChatExtensionConfiguration> GetDataSources(this CreateChatCompletionRequest request)
+    {
+        // TODO: How can we validate that this is being called in the right context,
```

Comment:
How can we throw an exception when using the wrong client?  Seems like MRW.Options is useful here.